src/core/bridge.ts

```diff
@@ -87,24 +97,47 @@ export abstract class AbstractBridge {
     }
   }
 
-  private fireRequest(url: string, body: Request | any, queryParams: any = {}): Axios.IPromise<any> {
+  // tslint:disable-next-line max-line-length
+  private fireRequest(url: string, body: Request | any, queryParams: any = {}): fetchPonyfill.Promise<any> {
     const options = {
-      url,
-      method: 'post',
-      params: queryParams,
-      data: this.augmentRequest(body),
-      headers: this.headers,
-      responseType: 'json',
-      timeout: this.config.timeout
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        ...this.headers
+      },
+      body: JSON.stringify(this.augmentRequest(body)),
     };
-    return axios(options)
-      .then((res) => res.data)
+    const timeoutPromise = () =>
```

Comment:
rather than make this a function created every time `fireRequest()` is called, let's make this a function at the top level called `createTimeoutPromise()` that accepts a `timeout` parameter